tests/unit/module_utils/test_elbv2.py

```diff
@@ -159,3 +161,133 @@ def test_get_elb_attributes(self):
+        [
+            (None, "None"),
+            (None, "HTTP1Only"),
+            ("HTTP1Only", "HTTP2Only"),
+        ],
+    )
+    def test__compare_listener_alpn_policy(self, current_protocol, current_alpn, new_alpn):
+        current_listener = self.createListener(protocol=current_protocol, alpnPolicy=[current_alpn])
+        new_listener = self.createListener(protocol="TLS", alpnPolicy=[new_alpn])
+        result = None
+        if current_protocol != "TLS":
+            result = {"Protocol": "TLS"}
+        if new_alpn and any((current_protocol != "TLS", not current_alpn, current_alpn and current_alpn != new_alpn)):
+            result = result or {}
+            result["AlpnPolicy"] = [new_alpn]
+
+        assert result == elbv2.ELBListeners._compare_listener(current_listener, new_listener)
+
+    @pytest.mark.parametrize(
+        "current_protocol,new_protocol",
+        [
+            ("TCP", "TCP"),
+            ("TLS", "HTTPS"),
+            ("HTTPS", "HTTPS"),
+            ("TLS", "TLS"),
+            ("HTTPS", "TLS"),
+            ("HTTPS", "TCP"),
+            ("TLS", "TCP"),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "current_ssl,new_ssl",
+        [
+            (None, "ELBSecurityPolicy-TLS-1-0-2015-04"),
+            ("ELBSecurityPolicy-TLS13-1-2-Ext2-2021-06", "ELBSecurityPolicy-TLS-1-0-2015-04"),
+            ("ELBSecurityPolicy-TLS-1-0-2015-04", None),
+            ("ELBSecurityPolicy-TLS-1-0-2015-04", "ELBSecurityPolicy-TLS-1-0-2015-04"),
+        ],
+    )
+    def test__compare_listener_sslpolicy(self, current_protocol, new_protocol, current_ssl, new_ssl):
+        current_listener = self.createListener(protocol=current_protocol, sslPolicy=current_ssl)
+
+        new_listener = self.createListener(protocol=new_protocol, sslPolicy=new_ssl)
+
+        expected = None
+        if new_protocol != current_protocol:
+            expected = {"Protocol": new_protocol}
+        if new_protocol in ("HTTPS", "TLS") and new_ssl and new_ssl != current_ssl:
+            expected = expected or {}
+            expected["SslPolicy"] = new_ssl
+        assert expected == elbv2.ELBListeners._compare_listener(current_listener, new_listener)
+
+    @pytest.mark.parametrize(
+        "current_protocol,new_protocol",
+        [
+            ("TCP", "TCP"),
+            ("TLS", "HTTPS"),
+            ("HTTPS", "HTTPS"),
+            ("TLS", "TLS"),
+            ("HTTPS", "TLS"),
+            ("HTTPS", "TCP"),
+            ("TLS", "TCP"),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "current_certificate,new_certificate",
+        [
+            (None, "arn:aws:iam::012345678901:server-certificate/ansible-test-1"),
+            (
+                "arn:aws:iam::012345678901:server-certificate/ansible-test-1",
+                "arn:aws:iam::012345678901:server-certificate/ansible-test-2",
+            ),
+            ("arn:aws:iam::012345678901:server-certificate/ansible-test-1", None),
```

Comment:
Same here, should we add a case where the certificates are the same?

Reply:
updated